js/client/modules/@arangodb/testutils/ldap-generic-failover.js

```diff
@@ -0,0 +1,147 @@
+/*jshint globalstrict:false, strict:false */
+/*global fail, assertTrue, assertEqual, fail */
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test the authentication (dual ldap)
+///
+/// @file
+///
+/// DISCLAIMER
+///
+/// Copyright 2021 ArangoDB GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is triAGENS GmbH, Cologne, Germany
+///
+/// @author Frank Celler
+/// @author Copyright 2021, triAGENS GmbH, Cologne, Germany
+////////////////////////////////////////////////////////////////////////////////
+
+const arango = require("@arangodb").arango;
+const db = require("internal").db;
+const users = require("@arangodb/users");
+
+// access forbidden to server/database
+const FORBIDDEN = exports.FORBIDDEN = 'FORBIDDEN';
+
+// access ok
+const OK = exports.OK = 'OK';
+
+// generate a single test
+const buildTestFunction = function(value) {
```

Comment:
This is ok, just FYI:

I tend to like to use object destructuring here

e.g.
`const buildTestFunction = function([user, pw, dynamo, result]) {`

or

`const buildTestFunction = function({user, pw, dynamo, result}) {`

This way the function signature is more expressive, and easier to figure out how to use.
I slightly prefer the object notation, because you are not bound to ordering of attributes, and you basically get "named parameters".